### PR TITLE
docs: clarify node 16 incompatibility

### PR DESCRIPTION
**Pull request by bensomers**
## Description
Clarify that node 16 currently does not work - running `make all` will throw errors around installing `libsass`.

## Test Plan

Docs change only.

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by hamidzr — APPROVED** [review]
🚢 
thanks Ben

**Comment by hamidzr** [comment]
on second thought maybe we should just advertise the latest LTS. 14 is the latest lts  https://nodejs.org/en/download/releases/

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ git clone git@github.com:determined-ai/determined.git
 
 - Go (>= 1.16)
 - Python (>= 3.6, < 3.8)
-- Node (>= 12)
+- Node (>= 12, < 16)
```
Comment (hamidzr) [inline CONTRIBUTING.md:53]: suggestion: I still have some unresolved questions with 15 let's stick with `<15`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ git clone git@github.com:determined-ai/determined.git
...
 - NPM (>= 6.12)
```
Comment (hamidzr) [inline CONTRIBUTING.md:54]: npm usually comes bundled with node I've never mixed and matched these manually but to be safe let's also put an upper limit here.

the latest active lts comes with `Node.js 14.17.1	Fermium	2021-06-15	8.4.371.23	NPM: 6.14.13	83`

Comment (bensomers) [reply CONTRIBUTING.md:54]: I don't know that we need to upper-bound NPM until we actually have evidence it doesn't work on newer releases.(though `6.14.13` is what I used here)

**Comment by shiyuann** [comment]
Does this PR accidentally come with a bunch of unrelated commits?

**Comment by bensomers** [comment]
> Does this PR accidentally come with a bunch of unrelated commits? @bensomers

it's just rebased on top of latest master

**Pull request closed without merging**